src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1482,6 +1496,45 @@ internal static bool ReportInvalidScopedOverrideAsError(MethodSymbol baseMethod,
                 return allowVariance && baseScope == ScopedKind.None;
             }
         }
+
+        /// <returns>
+        /// <see langword="true"/> if a diagnostic was added.
+        /// </returns>
+        internal static bool CheckRefReadonlyInMismatch<TArg>(
+            MethodSymbol? baseMethod,
+            MethodSymbol? overrideMethod,
+            BindingDiagnosticBag diagnostics,
+            ReportMismatchInParameterType<(ParameterSymbol BaseParameter, TArg Arg)> reportMismatchInParameterType,
+            TArg extraArgument,
+            // PROTOTYPE: This argument is expected to be used for method conversion warnings.
+            bool invokedAsExtensionMethod)
+        {
+            Debug.Assert(reportMismatchInParameterType is { });
+
+            if (baseMethod is null || overrideMethod is null)
+            {
+                return false;
+            }
+
+            bool hasErrors = false;
+            var baseParameters = baseMethod.Parameters;
+            var overrideParameters = overrideMethod.Parameters;
+            var overrideParameterOffset = invokedAsExtensionMethod ? 1 : 0;
+            Debug.Assert(baseMethod.ParameterCount == overrideMethod.ParameterCount - overrideParameterOffset);
+
+            for (int i = 0; i < baseParameters.Length; i++)
+            {
+                var baseParameter = baseParameters[i];
+                var overrideParameter = overrideParameters[i + overrideParameterOffset];
+                if ((baseParameter.RefKind, overrideParameter.RefKind) is (RefKind.RefReadOnlyParameter, RefKind.In) or (RefKind.In, RefKind.RefReadOnlyParameter))
+                {
+                    reportMismatchInParameterType(diagnostics, baseMethod, overrideMethod, overrideParameter, topLevel: true, (baseParameter, extraArgument));
+                    hasErrors = true;
+                }
+            }
+
+            return hasErrors;
```

Comment:
>return hasErrors; [](http://example.com/codeflow?start=12&length=17)

Do we use return value anywhere?

Reply:
Yes:

[https://github.com/dotnet/roslyn/blob/82f6a3df39e5375064551ed681465840d2a31995/src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs#L568-L581](https://github.com/dotnet/roslyn/blob/82f6a3df39e5375064551ed681465840d2a31995/src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs#L568-L581)